lib/facilities/local_store.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+require 'facilities/bulk_client'
+require 'facilities/point'
+
+module Facilities
+  class LocalStore
+    attr_reader :adapter
+
+    GIS_CHECK_FREQUENCY = 1.hour
+
+    def initialize(url, _type, adapter)
+      @url = url
+      @adapter = adapter
+      @mutex = Mutex.new
+      @last_check = Time.new(0).utc
+      @current_gis_update = 0
+      @client = BulkClient.new(@url)
+      @index = {}
+      @coordinates = []
+    end
+
+    def get(id)
+      check_freshness_and_update
+      @mutex.synchronize do
+        return @index[id]
+      end
+    end
+
+    def query(bbox, services = nil)
+      check_freshness_and_update
+      @mutex.synchronize do
+        result = @coordinates.select { |p| p.within?(bbox) }
+        result = result.sort_by { |p| p.dist_from_center(bbox) }
+        result = result.map(&:facility)
+        if services
+          result = result.select { |f| @adapter.services?(f, services) }
+        end
+        result
+      end
+    end
+
+    def check_freshness_and_update
```

Comment:
Right now they are changing ~ weekly as we are introducing new aspects of facilities and the administrations are actively cleaning their data.  Longer term it's probably going to be more like monthly or quarterly as administrations do a refresh of their data and it gets pushed to GIS.

I considered and started implementing a Redis approach. I discarded it, mainly for the same reason that Carl's nginx stale caching idea didn't seem right - the typical use is not getting individual resources by ID, it's doing a bounding box query. So even if the full records are stored in redis, I have to store the coordinates and IDs locally, and now I have two levels of caching which seemed over-engineered for this application. 

To be clear, we are checking the GIS data timestamp once an hour, and only pulling down the full dataset if the timestamp indicates that there is new data.